### PR TITLE
add srt support

### PR DESCRIPTION
**Pull request by vxzhong**
增加srt字幕的支持。支持 --accumulated_num参数。

### CODE REVIEW

**Comment by yihong0618** [comment]
@vxzhong can you add a test srt?

**Review by yihong0618 — COMMENTED** [review]
seems this is not support google translate and other translator?

**Comment by vxzhong** [comment]
In the early days, I tested with Google, but later switched to OpenAI.  Let me check.

**Comment by vxzhong** [comment]
I can run it with "python3 make_book.py --book_name test_books/Lex_Fridman_episode_322.srt --model google". From my understading, the loader and the translator are not related each other.

**Comment by yihong0618** [comment]
got it, will test tonight.
Thanks

**Comment by vxzhong** [comment]
I find the problem: quantifiers were added by Google Translate to the numbers, so the int() throw a exception. I try to fix this

File: book_maker/loader/srt_loader.py [hunk 1]
```diff
@@ -1 +1,325 @@
...
+
+                            if not self._check_blocks(
+                                translated_blocks, self.blocks[begin:end]
+                            ):
+                                raise Exception(
+                                    f"retry failed, adjust the srt manually."
+                                )
+
+                    i = 0
+                    for block in translated_blocks:
```
Comment (yihong0618) [inline book_maker/loader/srt_loader.py:254]: why not enumerate here?

File: book_maker/loader/srt_loader.py [hunk 2]
```diff
@@ -1 +1,325 @@
...
+                        text = block.get("text", "")
+                        self.p_to_save.append(text)
+                        self.bilingual_result.append(
+                            f"{self._get_block_text(self.blocks[begin + i])}\n{text}"
+                        )
+                        i += 1
+                else:
+                    i = 0
+                    for block in self.blocks[begin:end]:
```
Comment (yihong0618) [inline book_maker/loader/srt_loader.py:263]: ditto

**Review by yihong0618 — COMMENTED** [review]
thanks quite neat code!

File: book_maker/loader/srt_loader.py (hunk 2 above)
Comment (vxzhong) [reply book_maker/loader/srt_loader.py:263]: Let me fix this, not quite familiar with the python way. 😓

**Comment by yihong0618** [comment]
@zstone12 can you help to take a look, when you have time.

**Comment by zstone12** [comment]
> @zstone12 can you help to take a look, when you have time.

Ok.I will take a look at it a little later.

File: book_maker/loader/srt_loader.py [hunk 3]
```diff
@@ -1 +1,279 @@
...
+inspired by: https://github.com/jesselau76/srt-gpt-translator, MIT License
+"""
+import re
+import sys
+from pathlib import Path
+
+from book_maker.utils import prompt_config_to_kwargs
+
+from .base_loader import BaseBookLoader
+from book_maker.translator import ChatGPTAPI
```
Comment (zstone12) [inline book_maker/loader/srt_loader.py:11]: Can you delete this unused import?

Comment (vxzhong) [reply book_maker/loader/srt_loader.py]: done

**Comment by zstone12** [comment]
@vxzhong  This code looks good to me. And can you add some docs and ci test to `make_test_ebook.yaml` for it?
(By the way, Lex Fridman Podcast is pretty interesting.

**Comment by yihong0618** [comment]
@zstone12 @vxzhong  thank you very much
I will add the test add doc

**Pull request merged**